### PR TITLE
Add pre-commit bash script, add storybook tests to pre-commit

### PR DESCRIPTION
**Pull request by patthiel**
* Adds a pre-commit script that runs `yarn lint`, `yarn test` and selectively runs `yarn storybook:e2e` (determined by if files in your commit match `src/components/`)

* Autostart selenium-standalone to run storybook:e2e tests (thanks to a service plugin)

# To Test

```bash
yarn storybook
yarn storybook:e2e # in a separate shell
```
selenium should auto start and run the tests!

# To Test pre-commit hook:
1. Make a change to a file in `src/components`
2. Commit this change
3. Observe, lint, test and storybook:e2e run!
4. Make a change to files not in `src/components`
5. Commit this change
6. Observe, only lint and test run

🎆 🍾 🍾 🎈 🎉 🎉 🌮

### CODE REVIEW

**Comment by patthiel** [comment]
Neglected to mention that you'll need to run `yarn` to install the newly added dependency. That's most likely the cause for your errors. 

We could make it faster by having it run in headless chrome and running in parallel to the unit tests and linting. I will make the change in the morning to use headless and see how that affects the speed. An alternative I see is that we make github build checks more accurate (the builds have been failing since may 1 in baker), yet no one has noticed. We could also make it report out in one of our channels that the builds are failing (also not currently happening).

**Comment by rjstires** [comment]
@patthiel note at the top of the script I run `yarn && yarn storybook:e2e`

**Comment by patthiel** [comment]
let's debug tomorrow, but i think this may be your issue: https://github.com/Codeception/Codeception/issues/4657#issuecomment-348354476 , something about an additional rogue java installation

**Comment by rjstires** [comment]
```
─ yarn storybook:e2e
yarn run v1.3.2
$ ./node_modules/.bin/wdio ./e2e/config/wdio.storybook.conf.js

․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․F․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․․

120 passing (32.30s)
1 failing

1) Error State Component Suitesuite1 should display error state with text elements:
Failed: Promise was rejected with the following reason: timeout
running chrome
Error: Promise was rejected with the following reason: timeout
    at element("#storybook-preview-iframe") - index.js:316:3



error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

**Comment by rjstires** [comment]
Got it up and running, thanks for the link. I rolled back to Java 8 and it's running just fine.

**Comment by patthiel** [comment]
@asauber @rjstires this is ready for review 👨‍💻

File: src/components/ErrorState/ErrorState.spec.js [hunk 1]
```diff
@@ -22,7 +22,7 @@ describe('Error State Component Suite', () => {
         browser.click(navChild);
 
         waitForFocus('[data-qa-error-icon]');
-
+        console.log('balasdfg');
```
Comment (asauber) [inline src/components/ErrorState/ErrorState.spec.js]: let's remove this

File: scripts/pre-commit.sh [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+testStatus=$( cat .tmp.test.status )
+
+if [[ -z ${storybookStatus+x} ]]; then
+storybookStatus=$( cat .tmp.storybook.status )
+fi
+
+status=$(($lintStatus + $testStatus + $storybookStatus))
+
+# Remove temp files
+$( rm -rf .tmp.lint.status .tmp.test.status .tmp.storybook.status)
```
Comment (asauber) [inline scripts/pre-commit.sh]: let's add .tmp*.status to .gitignore

File: src/components/ErrorState/ErrorState.spec.js (hunk 1 above)
Comment (patthiel) [reply src/components/ErrorState/ErrorState.spec.js]: oops! bad commit, thanks

File: scripts/pre-commit.sh (hunk 2 above)
Comment (patthiel) [reply scripts/pre-commit.sh]: 👍

**Pull request merged**